sympy/solvers/ode/tests/test_systems.py

```diff
@@ -0,0 +1,457 @@
+    #a, b, c, d, e, f = symbols('a b c d e f')
+    #A = Matrix([
+    #[-a,  b,          c,  d],
+    #[ a, -b,          e,  0],
+    #[ 0,  0, -c - e - f,  0],
+    #[ 0,  0,          f, -d]])
+
+    A = Matrix([[0, I], [I, 0]])
+    expAt = Matrix([
+    [exp(I*t)/2 + exp(-I*t)/2, exp(I*t)/2 - exp(-I*t)/2],
+    [exp(I*t)/2 - exp(-I*t)/2, exp(I*t)/2 + exp(-I*t)/2]])
+    assert matrix_exp(A, t) == expAt
+
+
+def test_sysode_linear_neq_order1():
+
+   x, y = symbols('x y', cls=Function)
+   a, b, c, t = symbols('a b c t')
+
+   # The 2x2 systems tested below are all handled by _linear_2eq_order1_type1
+   # rather than _linear_neq_order1_type1. I'm adding the tests here because
+   # I think that _linear_2eq_order1_type1 should be removed in favour of
+   # _linear_neq_order1_type1.
+
+   eq1 = [Eq(x(t).diff(t), x(t)), Eq(y(t).diff(t), y(t))]
+   sol1 = [Eq(x(t), C1*exp(t)), Eq(y(t), C2*exp(t))]
+   assert dsolve(eq1) == sol1
+   assert checksysodesol(eq1, sol1) == (True, [0, 0])
+
+   eq2 = [Eq(x(t).diff(t), 2*x(t)), Eq(y(t).diff(t), 3*y(t))]
+   #sol2 = [Eq(x(t), C1*exp(2*t)), Eq(y(t), C2*exp(3*t))]
+   sol2 = [Eq(x(t), C1*exp(2*t)), Eq(y(t), C2*exp(3*t))]
+   assert dsolve(eq2) == sol2
+   assert checksysodesol(eq2, sol2) == (True, [0, 0])
+
+   eq3 = [Eq(x(t).diff(t), a*x(t)), Eq(y(t).diff(t), a*y(t))]
+   sol3 = [Eq(x(t), C1*exp(a*t)), Eq(y(t), C2*exp(a*t))]
+   assert dsolve(eq3) == sol3
+   assert checksysodesol(eq3, sol3) == (True, [0, 0])
+
+   # Regression test case for issue #15474
+   # https://github.com/sympy/sympy/issues/15474
+   eq4 = [Eq(x(t).diff(t), a*x(t)), Eq(y(t).diff(t), b*y(t))]
+   sol4 = [Eq(x(t), C1*exp(a*t)), Eq(y(t), C2*exp(b*t))]
+   assert dsolve(eq4) == sol4
+   assert checksysodesol(eq4, sol4) == (True, [0, 0])
+
+   eq5 = [Eq(x(t).diff(t), -y(t)), Eq(y(t).diff(t), x(t))]
+   #sol5 = [Eq(x(t), C1*cos(t) - C2*sin(t)), Eq(y(t), C1*sin(t) + C2*cos(t))]
+   sol5 = [Eq(x(t), -C1*sin(t) - C2*cos(t)), Eq(y(t), C1*cos(t) - C2*sin(t))]
+   assert dsolve(eq5) == sol5
+   assert checksysodesol(eq5, sol5) == (True, [0, 0])
+
+   eq6 = [Eq(x(t).diff(t), -2*y(t)), Eq(y(t).diff(t), 2*x(t))]
+   #sol6 = [Eq(x(t), C1*cos(2*t) + C2*sin(2*t)), Eq(y(t), C1*sin(2*t) - C2*cos(2*t))]
+   sol6 = [Eq(x(t), -C1*sin(2*t) - C2*cos(2*t)), Eq(y(t), C1*cos(2*t) - C2*sin(2*t))]
+   assert dsolve(eq6) == sol6
+   assert checksysodesol(eq6, sol6) == (True, [0, 0])
+
+   eq7 = [Eq(x(t).diff(t), I*y(t)), Eq(y(t).diff(t), I*x(t))]
+   #sol7 = [Eq(x(t), C1*exp(I*t) + C2*exp(-I*t)), Eq(y(t), C1*exp(I*t) - C2*exp(-I*t))]
+   sol7 = [Eq(x(t), -C1*exp(-I*t) + C2*exp(I*t)), Eq(y(t), C1*exp(-I*t) + C2*exp(I*t))]
+   assert dsolve(eq7) == sol7
+   assert checksysodesol(eq7, sol7) == (True, [0, 0])
+
+   eq8 = [Eq(x(t).diff(t), -a*y(t)), Eq(y(t).diff(t), a*x(t))]
+   sol8 = [Eq(x(t), -I*C1*exp(-I*a*t) + I*C2*exp(I*a*t)), Eq(y(t), C1*exp(-I*a*t) + C2*exp(I*a*t))]
+   assert dsolve(eq8) == sol8
```

Comment:
I think we can make this use `sin/cos` by checking for conjugates instead of `im(e).is_nonzero`.

Reply:
Ok I will update the function and see if that works.